Teapot/app/src/main/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (C)  The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+cmake_minimum_required(VERSION 3.4.1)
+
+set(CMAKE_VERBOSE_MAKEFILE on)
+
+# build native_app_glue as a static lib
+include_directories(${ANDROID_NDK}/sources/android/native_app_glue)
+add_library(app-glue STATIC
+            ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
+
+# now build app's shared lib
+include_directories(./ndk_helper
+                    ${ANDROID_NDK}/sources/android/cpufeatures)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
+file(GLOB_RECURSE teapot_SRCS ./*.c*)
```

Comment:
This is actually not recommended in CMake, because if you add a new file here, CMake wouldn't realize it.